commands/command.go

```diff
@@ -209,6 +213,14 @@ func (c *BaseCommand) run(args []string) error {
 		c.storage = storage.New(fileStrategy)
 	}
 
+	user, err := c.User()
+	if err != nil {
+		return err
+	}
+
+	c.Service.SetFields(c.flagTelemetryOn, user.PublicAPIKey, c.Name)
```

Comment:
rather than a catch-all `SetFields` method here, lets break it out into proper setters:
```go
c.TelemetryService.SetCommand(c.Name)
c.TelemetryService.SetUserID(user.PublicAPIKey)
```

also, make sure to call `SetCommand` prior to attempting to read `c.User()`, just so we get as much info as possible (I think moving `SetCommand` to be the very first line of code in this `run` func might be best)